core/src/main/scala/org/apache/spark/shuffle/sort/SortShuffleManager.scala

```diff
@@ -131,7 +131,7 @@ private[spark] class SortShuffleManager(conf: SparkConf) extends ShuffleManager
       metrics: ShuffleReadMetricsReporter): ShuffleReader[K, C] = {
     val baseShuffleHandle = handle.asInstanceOf[BaseShuffleHandle[K, _, C]]
     val (blocksByAddress, canEnableBatchFetch) =
-      if (baseShuffleHandle.dependency.shuffleMergeEnabled) {
+      if (baseShuffleHandle.dependency.isShuffleMergeFinalizedMarked) {
```

Comment:
Given the changes, should this be `shuffleMergeEnabled && isShuffleMergeFinalizedMarked ` ?

Reply:
@mridulm For the retry case, `shuffleMergeEnabled` would be false but `isShuffleMergeFinalizedMarked` but we still want this to return true right to fetch merge blocks generated by the first attempt? Therefore, won't `isShuffleMergeFinalizedMarked` be enough? Basically as long as shuffle merge is finalized (doesn't matter if shuffle merge is allowed or not), check for merge blocks to be fetched.

Reply:
Makes sense